### DISCUSSION TITLE
Social: New setting for local font folder

### DISCUSSION THREAD

**Discussion by thorstenpeter-sophos** [squidfunk/mkdocs-material#7127] [Ideas]
Hello,

We need to create social cards that use our corporate font, which isn't available on Google Fonts. As a workaround, we currently copy the .ttf files for our font variants to `.cache/plugin/social/font` before running a build.

I'd suggest a plugin setting to specify a local font folder, with a fallback to Google Fonts if a font used by the template doesn't exist locally.

For example:

```
plugins:
    - social:
        font_dir: my/custom/dir
```

I'm happy to open a change request if you think this could be useful for others as well.

**Comment by alexvoss** [comment]
This came up a little while ago when Google made changes that broke the automated font downloading. The suggestion to add a font directory did raise concerns about the requirement to put the right font variations in the right place and with the right name, something that is done automatically for the fonts that are downloaded from Google.; also concerns about non-repeatable builds but I see fonts as just another asset, like images, that would be checked into a repo.

I could imagine a solution where fonts are staged into the `.cache` directory just like the downloaded fonts are, just from any font file in the directory pointed at. That way we would relieve the user of the need to provide fonts in just the right directory structure and with just the right filenames.

**Comment by kamilkrzyskow** [comment]
I believe this was answered in:
- https://github.com/squidfunk/mkdocs-material/issues/6747

but like Alex has said, it was before the Google Fonts API change, so perhaps squidfunks stance has changed on it, but probably not 🤔
